torch/_tensor.py

```diff
@@ -76,8 +76,10 @@ def __deepcopy__(self, memo):
                             self.q_per_channel_axis()
                     else:
                         raise RuntimeError(f"Unsupported qscheme {self.qscheme()} in deepcopy")
+                    # TODO: Once we decide to break serialization FC, no longer
+                    # need to wrap with TypedStorage
                     new_tensor = torch._utils._rebuild_qtensor(
-                        new_storage,
+                        torch.storage.TypedStorage(new_storage, self.dtype),
```

Comment:
NB: to break serialization FC, we also need to introduce code to understand the non-TypedStorage wrapped format prior to emitting it